docs/ref/csp.txt

```diff
@@ -0,0 +1,279 @@
+class-based views<decorating-class-based-views>`.
+
+.. function:: csp_exempt(enforced=True, report_only=True)(view)
+
+    This decorator marks a view as being exempt from the Content Security Policy
+    protection ensured by the middleware. Example:
+
+    .. code-block:: python
+
+        from django.http import HttpResponse
+        from django.views.decorators.csp import csp_exempt
+
+
+        @csp_exempt
+        def my_view(request):
+            return HttpResponse("This view will not have either CSP headers")
+
+    This decorator takes optional boolean arguments ``enforced`` and ``report_only`` to
+    customize which header is exempt. By default, both headers are exempt. Example:
+
+    .. code-block:: python
+
+        from django.http import HttpResponse
+        from django.views.decorators.csp import csp_exempt
+
+
+        @csp_exempt(report_only=False)
+        def my_view(request):
+            return HttpResponse("This view will not have a report-only CSP header")
+
+
+        @csp_exempt(enforced=False)
+        def another_view(request):
+            return HttpResponse("This view will not have an enforced CSP header")
+
+
+.. function:: csp_override(config, enforced=True, report_only=True)(view)
+
+    This decorator allows you to override the Content Security Policy for a specific view.
+    It takes a dictionary as an argument, similar to the ``SECURE_CSP`` and ``SECURE_CSP_REPORT_ONLY``
+    settings. This decorator also takes optional boolean arguments ``enforced`` and ``report_only``
+    to customize which header is overridden. By default, both headers are overridden. Example:
+
+    .. code-block:: python
+
+        from django.http import HttpResponse
+        from django.middleware import csp
+        from django.views.decorators.csp import csp_override
+
+
+        @csp_override(
+            {
+                "default-src": [csp.SELF],
+                "img-src": [csp.SELF, "data:"],
+            }
+        )
+        def my_view(request):
+            return HttpResponse("Both CSP headers will be overridden")
+
+
+        @csp_override(
+            {
+                "default-src": [csp.SELF],
+                "img-src": [csp.SELF, "data:"],
+            },
+            report_only=False,
+        )
+        def enforced_only_view(request):
+            return HttpResponse("Only the enforced CSP header will be overridden")
+
+
+        @csp_override(
+            {
+                "default-src": [csp.SELF],
+                "img-src": [csp.SELF, "data:"],
+            },
+            enforced=False,
+        )
+        def report_only_view(request):
+            return HttpResponse("Only the report-only CSP header will be overridden")
+
+These decorators provide fine-grained control over the Content Security Policy
+on a per-view basis, allowing you to customize the policy as needed for specific
+parts of your application.
+
+Limitations and Considerations
```

Comment:
Question: `django-csp` has a warning about the potential impact of disabling CSP for a specific path prefix (or view, since that's not implemented right now). Should that warning be duplicated here? 

> Excluding any path on your site will eliminate the benefits of CSP everywhere on your site. The typical browser security model for JavaScript considers all paths alike. A Cross-Site Scripting flaw on, e.g., `excluded-page/` can therefore be leveraged to access everything on the same origin.

My understanding is that any page view which is exempt has the same impact as excluding a prefix?

Reply:
I added a bullet point about this and also reworked the text a bit. I also ordered the concerns from what I considered highest risk to lowest.